modules/core/include/opencv2/core/softfloat.hpp

```diff
@@ -0,0 +1,325 @@
+    round_min         = 2,
+    round_max         = 3,
+    round_near_maxMag = 4,
+    round_odd         = 5
+};
+//fixed to make softfloat code stateless
+const uint_fast8_t globalRoundingMode = round_near_even;
+
+/*----------------------------------------------------------------------------
+| Software floating-point exception flags.
+*----------------------------------------------------------------------------*/
+enum {
+    flag_inexact   =  1,
+    flag_underflow =  2,
+    flag_overflow  =  4,
+    flag_infinite  =  8,
+    flag_invalid   = 16
+};
+
+// Disabled to make softfloat code stateless
+// The user has to check values manually with *_isSignalingNaN() functions
+CV_INLINE void raiseFlags( uint_fast8_t /* flags */)
+{
+    //exceptionFlags |= flags;
+}
+
+/*----------------------------------------------------------------------------
+*----------------------------------------------------------------------------*/
+
+#define signF32UI( a ) (((uint32_t) (a)>>31) != 0)
+#define expF32UI( a ) ((int_fast16_t) ((a)>>23) & 0xFF)
+#define fracF32UI( a ) ((a) & 0x007FFFFF)
+#define packToF32UI( sign, exp, sig ) (((uint32_t) (sign)<<31) + ((uint32_t) (exp)<<23) + (sig))
+
+#define isNaNF32UI( a ) (((~(a) & 0x7F800000) == 0) && ((a) & 0x007FFFFF))
+
+/*----------------------------------------------------------------------------
+*----------------------------------------------------------------------------*/
+
+#define signF64UI( a ) (((uint64_t) (a)>>63) != 0)
+#define expF64UI( a ) ((int_fast16_t) ((a)>>52) & 0x7FF)
+#define fracF64UI( a ) ((a) & UINT64_C( 0x000FFFFFFFFFFFFF ))
+#define packToF64UI( sign, exp, sig ) ((uint64_t) (((uint_fast64_t) (sign)<<63) + ((uint_fast64_t) (exp)<<52) + (sig)))
+
+#define isNaNF64UI( a ) (((~(a) & UINT64_C( 0x7FF0000000000000 )) == 0) && ((a) & UINT64_C( 0x000FFFFFFFFFFFFF )))
+
+/*----------------------------------------------------------------------------
+*----------------------------------------------------------------------------*/
+
+struct softfloat32_t;
+struct softfloat64_t;
+
+struct CV_EXPORTS softfloat32_t
+{
+public:
+    softfloat32_t() { v = 0; }
+    softfloat32_t( const softfloat32_t& c) { v = c.v; }
+    softfloat32_t& operator=( const softfloat32_t& c )
+    {
+        if(&c != this) v = c.v;
+        return *this;
+    }
+    static const softfloat32_t fromRaw( const uint32_t a ) { softfloat32_t x; x.v = a; return x; }
+
+    softfloat32_t( const uint32_t );
+    softfloat32_t( const uint64_t );
+    softfloat32_t( const int32_t );
+    softfloat32_t( const int64_t );
+    softfloat32_t( const float a ) { Cv32suf s; s.f = a; v = s.u; }
+
+    uint_fast32_t toUI32_minMag( bool exact = false ) const;
+    uint_fast64_t toUI64_minMag( bool exact = false ) const;
+    int_fast32_t   toI32_minMag( bool exact = false ) const;
+    int_fast64_t   toI64_minMag( bool exact = false ) const;
+    uint_fast32_t toUI32( uint_fast8_t roundingMode = round_near_even, bool exact = false ) const;
```

Comment:
`operator int() const` and `operator float() const` create ambiguity when casting to int, which is needed, for example, for GTEST checks. The problem could be solved with explicit casts but they are not allowed until C++11.
That is why there will be no `operator int() const`. There will be separate `trunc()` function outside the classes.